mongomock/aggregate.py

```diff
@@ -314,7 +315,29 @@ def _handle_string_operator(self, operator, values):
             "Although '%s' is a valid string operator for the aggregation "
             'pipeline, it is currently not implemented  in Mongomock.' % operator)
 
-    def _handle_date_operator(self, operator, values):
+    def _handle_date_operator(self, operator, values):      
+        if operator == "$dateToString":
+            if isinstance(values, dict):
```

Comment:
In general, please reverse your conditions: the short one should come first so that you can exit quickly.

```py
if not isintance(values, dict):
   raise OperationFailure...
valid_values_keys = ["format", "date"]
if not all(k in values.keys() for k in valid_values_keys):
  raise OperationFailure...
```

That allows for smaller indentations.